framework/meta/src/generate_snippets/snippet_type_map.rs

```diff
@@ -0,0 +1,296 @@
+use std::collections::HashMap;
+
+const INNER_TYPE_SEPARATOR: char = '<';
+const INNER_TYPE_END: char = '>';
+pub static DEBUG_API_SUFFIX: &str = "<DebugApi>";
+pub static PLACEHOLDER_INPUT_TYPE_NAME: &str = "PlaceholderInput";
+// pub static PLACEHOLDER_OUTPUT_TYPE_NAME: &str = "PlaceholderOutput";
+
+#[derive(Clone, Default)]
+pub struct RustTypeString {
+    type_name: String,          // used for return types
+    default_value_expr: String, // used for arguments
+    contains_custom_types: bool,
+}
+
+impl RustTypeString {
+    pub fn get_type_name(&self) -> &str {
+        &self.type_name
+    }
+
+    pub fn get_default_value_expr(&self) -> &str {
+        if !self.contains_custom_types {
+            &self.default_value_expr
+        } else {
+            PLACEHOLDER_INPUT_TYPE_NAME
+        }
+    }
+}
+
+lazy_static! {
+    static ref ABI_TYPES_TO_RUST_TYPES_MAP: HashMap<&'static str, RustTypeString> = {
+        let mut m = HashMap::new();
+
+        m.insert(
```

Comment:
It might be worth trying to initialize from array for conciseness. Example from official docs:
```
use std::collections::HashMap;

let solar_distance = HashMap::from([
    ("Mercury", 0.4),
    ("Venus", 0.7),
    ("Earth", 1.0),
    ("Mars", 1.5),
]);
```

Reply:
Tried it, looks worse due to the fact that we use a struct here.